library/ssl_tls13_generic.c

```diff
@@ -1060,10 +1060,14 @@ static int ssl_tls13_prepare_finished_message( mbedtls_ssl_context *ssl )
                     sizeof( ssl->handshake->state_local.finished_out.digest ),
                     &ssl->handshake->state_local.finished_out.digest_len,
                     ssl->conf->endpoint );
-
+    if( ssl->conf->endpoint == MBEDTLS_SSL_IS_CLIENT )
+    {
+        mbedtls_platform_zeroize( &ssl->handshake->tls13_hs_secrets,
```

Comment:
I would prefer this to be inside `mbedtls_ssl_tls13_calculate_verify_data` because that's where we use for the last time the handshake secrets. In `mbedtls_ssl_tls13_calculate_verify_data` it would be something like:
```C
    if( from == MBEDTLS_SSL_IS_CLIENT )
    {                                         
        base_key = ssl->handshake->tls13_hs_secrets.client_handshake_traffic_secret;
        base_key_len = sizeof( ssl->handshake->tls13_hs_secrets.client_handshake_traffic_secret );
    }
    else
    {                                                                        
        base_key = ssl->handshake->tls13_hs_secrets.server_handshake_traffic_secret;
        base_key_len = sizeof( ssl->handshake->tls13_hs_secrets.server_handshake_traffic_secret );
    }                                                                       
    ret = ssl_tls13_calc_finished_core( md_type, base_key, transcript, dst );   
    if( ret != 0 )                                                              
        goto exit;
    mbedtls_platform_zeroize( base_key, base_key_len );                                                             
    *actual_len = md_size;
```

Reply:
Thanks Ronald. That's better choice